### ISSUE TITLE
Adding `hasOne` fragment to unsaved record?

### ISSUE THREAD

**Issue by chrisvariety** [adopted-ember-addons/ember-data-model-fragments#3]
Hi,

Happy Easter.

I'm trying to add a fragment to an unsaved record like so:

``` js
this.store.createRecord('comment', {user: {email: 'test@test.com'}})
```

This blows up with: `Error: Assertion Failed: You can only assign a 'user' fragment to this property`

OK, so I can create a fragment like so:

``` js
this.store.createRecord('comment', {
  user: this.store.createFragment('user', {email: 'a@b.com'})
})
```

This works, but blows up after save on this line: https://github.com/lytics/ember-data.model-fragments/blob/master/packages/model-fragments/lib/fragments/states.js#L63

`record` is undefined because the `_owner` of the fragment doesn't get set above. I can set it with `user.setProperties({_owner: comment, _name: 'user'});` but this feels wrong to me. 

TL;DR What is the best way to add a hasOne fragment to an unsaved record?

Thanks!!
